lib/logstash/inputs/kafka.rb

```diff
@@ -292,14 +321,23 @@ def thread_runner(logstash_queue, consumer)
                 end
                 event.remove("message")
               end
-              if @decorate_events
+              if @metadata_mode.include?(:record_props)
                 event.set("[@metadata][kafka][topic]", record.topic)
                 event.set("[@metadata][kafka][consumer_group]", @group_id)
                 event.set("[@metadata][kafka][partition]", record.partition)
                 event.set("[@metadata][kafka][offset]", record.offset)
                 event.set("[@metadata][kafka][key]", record.key)
                 event.set("[@metadata][kafka][timestamp]", record.timestamp)
               end
+              if @metadata_mode.include?(:headers)
+                for header in record.headers do
```

Comment:
nit: the ruby community [tends to avoid the `for X in Y` syntax](https://github.com/rubocop/ruby-style-guide#no-for-loops) because it leads to scoping surprises.

```suggestion
                record.headers.each do |header|
```